src/node/nodestate.h

```diff
@@ -1431,7 +1431,9 @@ namespace ccf
         rpcsessions,
         *network.tables->get<pbft::RequestsMap>(pbft::Tables::PBFT_REQUESTS),
         *network.tables->get<pbft::PrePreparesMap>(
-          pbft::Tables::PBFT_PRE_PREPARES));
+          pbft::Tables::PBFT_PRE_PREPARES),
+        node_kp->private_key_pem().str(),
```

Comment:
Same for the private key (`std::vector<uint8_t>`).

Reply:
I changed `cert` since it already is a `vector<uint8_t>` so there is not need to convert it into a string, but since the private key is already a string and we can use it to create a `keypair` I think it is fine to pass it in as a string